Sources/Plasma/PubUtilLib/plAudio/plAudioSystem.cpp

```diff
@@ -987,9 +984,78 @@ bool plAudioSystem::MsgReceive(plMessage* msg)
     return hsKeyedObject::MsgReceive(msg);
 }
 
+bool plAudioSystem::BeginCapture()
+{
+    if (fCaptureDevice) {
+        if (!EndCapture())
+            return false;
+    }
+
+    ALCsizei bufferSize = fCaptureFrequency * sizeof(int16_t) * BUFFER_LEN_SECONDS;
+    fCaptureDevice = alcCaptureOpenDevice(nullptr, fCaptureFrequency, AL_FORMAT_MONO16, bufferSize);
+    if (fCaptureDevice) {
+        alcCaptureStart(fCaptureDevice);
+        return true;
+    } else {
+        return false;
+    }
+}
+
+bool plAudioSystem::CaptureSamples(uint32_t samples, int16_t* data) const
+{
+    if (!fCaptureDevice)
+        return false;
+
+    alcCaptureSamples(fCaptureDevice, data, samples);
+    return true;
+}
+
+uint32_t plAudioSystem::GetCaptureSampleCount() const
+{
+    if (!fCaptureDevice)
+        return 0;
+
+    ALCsizei samples;
+    alcGetIntegerv(fCaptureDevice, ALC_CAPTURE_SAMPLES, sizeof(samples), &samples);
+    return samples;
+}
+
+bool plAudioSystem::SetCaptureSampleRate(uint32_t sampleRate)
+{
+    if (fCaptureFrequency != sampleRate) {
+        fCaptureFrequency = sampleRate;
+        if (fCaptureDevice) {
+            if (!EndCapture())
+                return false;
+            return BeginCapture();
+        }
+    }
+    return true;
+}
+
+bool plAudioSystem::EndCapture()
+{
+    if (!fCaptureDevice)
+        return false;
+
+    alcCaptureStop(fCaptureDevice);
+
+    // discard any samples not processed
+    ALCsizei samples;
+    alcGetIntegerv(fCaptureDevice, ALC_CAPTURE_SAMPLES, sizeof(samples), &samples);
+    if (samples) {
+        std::unique_ptr<int16_t> buf{ new int16_t[samples] };
```

Comment:
I believe we require C++14 by now, so this would be better as `std::make_unique<int16_t[]>(samples)`

Reply:
Is `std::make_unique` available in VC++2013?

Reply:
Hmm, good question.  Is there a reason we still support VC++2013 (6 years old now)?  And can we... not?

Reply:
I just looked it up in a huge list on the MS website, and it appears make_unique was added in VC++ 2013. Bumping the compiler requirement is probably not a bad idea at this stage, I just preferred for my PR to not depend on that 😄

Reply:
Ah, good enough for me :)